packages/patternfly-4/react-core/src/components/Nav/Nav.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** NavList elements */
+  children(props, propName, componentName) {
+    const prop = props[propName];
+
+    let error = null;
+    React.Children.forEach(prop, child => {
+      if (child.type.componentType !== 'NavList') {
+        error = new Error(`${componentName} children should be of type NavList.`);
+      }
+    });
+    return error;
+  },
+  /** Additional classes added to the Nav */
+  className: PropTypes.string,
+  /** Non-visible label to describe the nav */
+  ariaLabel: PropTypes.string
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  ariaLabel: ''
+};
+
+const renderLists = (children, updateActive, activeGroup, activeItem) =>
+  React.Children.map(children, (childNode, index) =>
+    React.cloneElement(childNode, {
+      groupId: index,
+      updateActive,
+      activeGroup,
+      activeItem
+    })
+  );
+
+class Nav extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeItem: -1,
+      activeGroup: -1
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const { activeItem, activeGroup } = this.state;
+    if (activeItem === nextState.activeItem && activeGroup === nextState.activeGroup) {
```

Comment:
Do I understand it correctly, that when consumers want to update the nav items they can only update if some item is active or not? What if there are some conditional items which will be shown only after some event happens? If so, is this correct behavior?

Reply:
Removing this